src/link/link_info/gre/gre_tun6.rs

```diff
@@ -0,0 +1,193 @@
+// SPDX-License-Identifier: MIT
+
+use std::net::Ipv6Addr;
+
+use crate::ip::parse_ipv6_addr;
+use anyhow::Context;
+use byteorder::{ByteOrder, NetworkEndian};
+use netlink_packet_utils::{
+    nla::{DefaultNla, Nla, NlaBuffer},
+    parsers::{parse_u16_be, parse_u32_be, parse_u8},
+    DecodeError, Parseable,
+};
+
+use super::{
+    GreEncapFlags, GreEncapType, GreIOFlags, IFLA_GRE_COLLECT_METADATA,
+    IFLA_GRE_ENCAP_DPORT, IFLA_GRE_ENCAP_FLAGS, IFLA_GRE_ENCAP_LIMIT,
+    IFLA_GRE_ENCAP_SPORT, IFLA_GRE_ENCAP_TYPE, IFLA_GRE_FLAGS,
+    IFLA_GRE_FLOWINFO, IFLA_GRE_FWMARK, IFLA_GRE_IFLAGS, IFLA_GRE_IKEY,
+    IFLA_GRE_LOCAL, IFLA_GRE_OFLAGS, IFLA_GRE_OKEY, IFLA_GRE_REMOTE,
+    IFLA_GRE_TTL, IFLA_GRE_UNSPEC,
+};
+
+pub use InfoGreTun6 as InfoGreTap6;
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+#[non_exhaustive]
+pub enum InfoGreTun6 {
+    Unspec,
+    //Link,
+    IFlags(GreIOFlags),
+    OFlags(GreIOFlags),
+    IKey(u32),
+    OKey(u32),
+    Local(Ipv6Addr),
+    Remote(Ipv6Addr),
+    Ttl(u8),
+    EncapLimit(u8),
+    FlowLabel(u32),
+    Flags(u32), // TODO: expose IP6_TNL_F_* flags
```

Comment:
If todo, then remove it. Otherwise, we have to break the API when we change it bitflags.

Reply:
Currently we break the API all the time by substituting generic Netlink attributes with concrete bindings like is done here. Deserializing will now yield the concrete enums instead of the generic Netlink attributes. 

I just haven't looked into what the flags are for in the Kernel yet and so I left them out.

I don't really get why we'd want to "remove" todos and why removing them would protect us from breaking the API in the long term if we'd still implement it eventually (since this is what a todos is, if present as a comment as a reminder or not).

Reply:
With `[non_exhaustive]`, adding entry to an enum or struct will not break API.